site/docs/v1/tech/events-webhooks/events.adoc

```diff
@@ -27,6 +27,24 @@ These are the events that FusionAuth generates that can be optionally consumed b
 * <<JWT Refresh>> - when an access token is refreshed using a refresh token
 * <<JWT Refresh Token Revoke>> - when a refresh token (or multiple tokens) are revoked
 
+=== Application Scoped Events
+
+Some events can be generated for one or more specified applications. These events may also be generated for all applications within a tenant.
+
+The events that are considered "application" events and can be scoped to an application based up the Webhook config: 
```

Comment:
For your own awareness, technically, "an application event" is any event that extends `ApplicationEvent`. 

Currently this is :

- `JWTPublicKeyUpdateEvent` (`jwt.public-key.update`)
- `JWTRefreshTokenRevokeEvent` (`jwt.refresh-token.revoke`)
- `UserActionEvent` (`user.action`)

Then, other events may have an `applicationId` in the event body which would include:

- `JWTRefreshEvent` (`jwt.refresh`)
- `UserLoginFailedEvent` (`user.login.failed`)
- `UserLoginSuccessEvent` (`user.login.success`)
- `UserRegistrationCreateEvent` (`user.registration.create`)
- `UserRegistrationDeleteEvent` (`user.registration.delete`)
- `UserRegistrationUpdateEvent` (`user.registration.update`)
- `UserRegistrationVerifiedEvent` (`user.registration.verified`)

So in summary, this is all sort of a mess. So when you "select an application" in the webhook configuration, this is now only going to ever get events that extend `ApplicationEvent` and if the event is for that application. So in almost all cases, with the current design, people will want to use "global" webhook configurations - and not select an application. 

This whole system needs to be re-worked.

Reply:
I saw that superclass and even considered linking to the java client lib so that we wouldn't need to update the list. 

I think maybe a better way forward would be to warn folks that application specific events are legacy and due for a rework (as you mention in your other comment). I'll also update to mention that tenant scoped events are almost certainly what you want.